crates/theme/src/registry.rs

```diff
@@ -26,40 +30,41 @@ pub struct ThemeMeta {
 ///
 /// This should not be exposed outside of this module.
 #[derive(Default, Deref, DerefMut)]
-struct GlobalThemeRegistry(ThemeRegistry);
+struct GlobalThemeRegistry(Arc<ThemeRegistry>);
 
 /// Initializes the theme registry.
 pub fn init(assets: Box<dyn AssetSource>, cx: &mut AppContext) {
-    cx.set_global(GlobalThemeRegistry(ThemeRegistry::new(assets)));
+    cx.set_global(GlobalThemeRegistry(Arc::new(ThemeRegistry::new(assets))));
+}
+
+struct ThemeRegistryState {
+    themes: HashMap<SharedString, Arc<Theme>>,
 }
 
 pub struct ThemeRegistry {
+    state: RwLock<ThemeRegistryState>,
```

Comment:
Now that the `ThemeRegistry` is held in an `Arc`, we need to rely on interior mutability to make changes to the registry.

We're using the same pattern here that we use in the `LanguageRegistry`.